src/main.cpp

```diff
@@ -3,21 +3,45 @@
 #include "datalogging.h"
 #include "gnss.h"
 #include "sensing.h"
+#include "types.h"
+
+bool DEBUG = true;
+
+static SensingData data;
+
+static void printSensingData() {
+    Serial.print("MS5607 Altitude: ");
+    Serial.print(data.baroAltitude);
+    Serial.print(" m, ");
+    Serial.print("Pressure: ");
+    Serial.print(data.pressure);
+    Serial.print(" mbar, ");
+    Serial.print("Temp: ");
+    Serial.print(data.temperature);
+    Serial.println(" C");
+
+    Serial.print("HDC2080 Temp: ");
+    Serial.print(data.hdcTemperature);
+    Serial.print(" C, ");
+    Serial.print("Humidity: ");
+    Serial.print(data.humidity);
+    Serial.println(" %");
+}
 
 void setup() {
     Serial.begin(115200);
-    delay(5000);
+    delay(3000);
 
     sensingInit();
     dataloggingInit();
     gnssInit();
-
-    // xTaskCreate(sensingTask, "Sensing Task", 8192, NULL, 1, NULL);
-    // xTaskCreate(dataloggingTask, "Datalogging Task", 8192, NULL, 1, NULL);
-    // xTaskCreate(gnssTask, "GNSS Task", 8192, NULL, 1, NULL);
 }
 
 void loop() {
-    Serial.println("Main loop running");
-    sensingExecute();
+    sensingExecute(data);
```

Comment:
i assume this is just bc its partway through, but is the idea to superloop here? or real threads as sensingTask would seem to indicate